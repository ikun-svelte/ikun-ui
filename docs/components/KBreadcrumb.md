---
title: KBreadcrumb
lang: en-US
---

# KBreadcrumb

Displays the location of the current page, making it easier to browser back.

## Install

::: code-group

```bash [pnpm]
pnpm add @ikun-ui/breadcrumb @ikun-ui/breadcrumb-item
```

```bash [yarn]
yarn add @ikun-ui/breadcrumb @ikun-ui/breadcrumb-item
```

```bash [npm]
npm install @ikun-ui/breadcrumb @ikun-ui/breadcrumb-item
```

:::

## Basic usage

In `KBreadcrumb`, each `KBreadcrumbItem` is a tag that stands for every level starting from homepage. This component has a String attribute `separator`, and it determines the `separator`. Its default value is '/'.

<demo src="breadcrumb/basic.svelte"  github='Breadcrumb'></demo>

## Icon separator

Set `separatorIcon` to use `KIcon` as the separator, it will cover `separator`.

<demo src="breadcrumb/separator-icon.svelte"  github='Breadcrumb'></demo>

## Breadcrumb Props

| Name          | Type                     | Default | Description                                                |
| ------------- | ------------------------ | ------- | ---------------------------------------------------------- |
| separator     | `string`                 | `/`     | separator character                                        |
| separatorIcon | `string`                 | `''`    | The class name of the icon, following the unocss standard. |
| cls           | `string`                 | `''`    | Additional class for component                             |
| attrs         | `Record<string, string>` | `{}`    | Additional attributes                                      |

## Breadcrumb Slots

| Name    | Description               |
| ------- | ------------------------- |
| default | Customize default content |

## BreadcrumbItem Props

| Name  | Type                     | Default | Description                                                                                                                                                |
| ----- | ------------------------ | ------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
| href  | `string`                 | `''`    | The URL that the hyperlink points to, same as [href <span class="i-carbon-link text-12px" />](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) |
| cls   | `string`                 | `''`    | Additional class for component                                                                                                                             |
| attrs | `Record<string, string>` | `{}`    | Additional attributes                                                                                                                                      |

## BreadcrumbItem Slots

| Name    | Description               |
| ------- | ------------------------- |
| default | Customize default content |
