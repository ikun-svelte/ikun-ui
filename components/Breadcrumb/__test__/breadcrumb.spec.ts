import { afterEach, expect, test, describe, beforeEach } from 'vitest';
import KBreadcrumb from '../src';

let host: HTMLElement;

const initHost = () => {
	host = document.createElement('div');
	host.setAttribute('id', 'host');
	document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
});
afterEach(() => {
	host.remove();
});

describe('Test: KBreadcrumb', () => {
	test('props: cls', async () => {
		const instance = new KBreadcrumb({
			target: host,
			props: {
				cls: 'k-breadcrumb--test'
			}
		});
		expect(instance).toBeTruthy();
		expect((host as HTMLElement)!.innerHTML.includes('k-breadcrumb--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
