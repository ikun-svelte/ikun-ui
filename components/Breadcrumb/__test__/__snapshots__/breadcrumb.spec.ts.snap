// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KBreadcrumb > props: cls 1`] = `"<div class=\\"k-breadcrumb k-breadcrumb--test\\"></div>"`;
